packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/summary.js

```diff
@@ -27,7 +27,7 @@ const Summary = props => {
         selectedPaymentMethod
     } = talonProps;
 
-    if (isLoading) {
+    if (isLoading || !selectedPaymentMethod) {
```

Comment:
What's the intent behind this change?

Reply:
Safety - the talon can return `null` for data but not be loading. In general we should check for truthiness of data rather than use the loading boolean going forward - this is a conversation I've been having in several places, now [here](https://github.com/magento/pwa-studio/issues/2635).

Reply:
Sure. I'm just uncertain about rendering a loading indicator when `loading` is false, even if `data` is falsy, because I believe `loading: true` is the only assurance we can have that `data` will be truthy in the future. Rather than potentially showing a loading indicator forever (which would be the worst), wouldn't we prefer to render empty text (or nothing at all)?

```es6
if (loading) return <Loading />
if (error) return <Error />
if (!data) return <Empty />

return <Content data={data} />
```

Reply:
> wouldn't we prefer to render empty text (or nothing at all)?

Yes, I think this is the key. I think we should be intentional about where we render loading spinners/states (page level?). I'd prefer we do away with most of the small loading spinners like in the account trigger or in each of these summary views.

For the specific line of code in question I think a simple change to `if (isLoading && !selectedPaymentMethod)` would suffice. "Only show the indicator if we are loading and don't have data."